pkg/scheduler/framework/plugins/podtopologyspread/plugin.go

```diff
@@ -141,9 +145,137 @@ func (pl *PodTopologySpread) EventsToRegister() []framework.ClusterEventWithHint
 		// an unschedulable Pod schedulable.
 		// - Delete. An unschedulable Pod may fail due to violating an existing Pod's topology spread constraints,
 		// deleting an existing Pod may make it schedulable.
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.All}},
-		// Node add|delete|updateLabel maybe lead an topology key changed,
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.All}, QueueingHintFn: pl.isSchedulableAfterPodChange},
+		// Node add|delete|update maybe lead an topology key changed,
 		// and make these pod in scheduling schedulable or unschedulable.
-		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Delete | framework.UpdateNodeLabel}},
+		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Delete | framework.Update}, QueueingHintFn: pl.isSchedulableAfterNodeChange},
```

Comment:
do we need to add a TODO here  to change `Update` to `UpdateNodeLabel` again, after https://github.com/kubernetes/kubernetes/issues/122306 fixed?

Reply:
Well, either is actually fine though, let's just keep it as it is, since we already have a comment in isSchedulableAfterNodeChange. If isSchedulableAfterNodeChange is re-implemented properly after https://github.com/kubernetes/kubernetes/issues/122306, an implementor or reviewer should just notice that they can change framework.Update to framework.UpdateNodeLabel.